packages/core-kernel/src/utils/is-blacklisted.ts

```diff
@@ -0,0 +1,18 @@
+import nm from "nanomatch";
+
+// todo: review the implementation
+export const isBlacklisted = (whitelist: string[], remoteAddress: string): boolean => {
```

Comment:
Is it exactly the same code as `isWhitelisted` ? If it is can you instead make a generic one like `matchesIPAddressList` ?

Reply:
No it is not exactly the same. IsBlackilsted is returning false by default if no list is provided, but isWhitelisted return true. I am aware of the duplicated for loop in both files, this can potentially be extracted.

Reply:
ah ok sorry no don't worry we can keep it this way but then can you change `whitelist` parameter to `blacklist` ? :sweat_smile:

Reply:
Of course. I have mislooked it.